salt/cloud/clouds/vmware.py

```diff
@@ -3041,8 +3038,11 @@ def create_snapshot(name, kwargs=None, call=None):
             'You must specify snapshot name for the snapshot to be created.'
         )
 
-    memdump = _str_to_bool(kwargs.get('memdump', True))
-    quiesce = _str_to_bool(kwargs.get('quiesce', False))
+    memdump = kwargs.get('memdump') if kwargs and 'memdump' in kwargs else True
```

Comment:
Lines 3041 to 3045 is not needed at all. 

If ``kwargs`` is None, program flow would exit above at https://github.com/saltstack/salt/pull/31011/files#diff-0fa55dcf4489823e04ae1ee95c844921R3037

If ``kwargs`` is not ``None`` and doesn't contain ``memdump`` or ``quiesce`` then ``kwargs.get`` would use the default passed to it which would be either ``True`` or ``False``